### PR TITLE
feature: voice channel support

### PR DESCRIPTION
**Pull request by knokbak**

### CODE REVIEW

File: src/classes/VoiceChannel.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+     * 
+     * @param {any} data The data of this channel.
+     */
+    private build (data: any): VoiceChannel {
+        if ('bitrate' in data) {
+            this.bitrate = data.bitrate;
+        }
+        if ('user_limit' in data) {
+            this.userLimit = data.user_limit || null;
+        }
```
Comment (axisiscool) [inline src/classes/VoiceChannel.ts]:
```suggestion
        if ('user_limit' in data) this.userLimit = data.user_limit ?? null;
```

File: src/classes/VoiceChannel.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+    /**
+     * Build the VoiceChannel.
+     * 
+     * @param {any} data The data of this channel.
+     */
+    private build (data: any): VoiceChannel {
+        if ('bitrate' in data) {
+            this.bitrate = data.bitrate;
+        }
```
Comment (axisiscool) [inline src/classes/VoiceChannel.ts]:
```suggestion
        if ('bitrate' in data) this.bitrate = data.bitrate;
```

File: src/classes/VoiceState.ts [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        }
+        if ('self_mute' in data) {
+            this.selfMute = data.self_mute;
+        }
+        if ('self_stream' in data) {
+            this.streaming = data.self_stream;
+        }
+        if ('self_video' in data) {
+            this.videoEnabled = data.self_video;
+        }
```
Comment (axisiscool) [inline src/classes/VoiceState.ts]:
```suggestion
        if ('session_id' in data) this.sessionId = data.session_id;
        if ('deaf' in data) this.deaf = data.deaf;
        if ('mute' in data) this.mute = data.mute;
        if ('self_deaf' in data) this.selfDeaf = data.self_deaf;
        if ('self_mute' in data) this.selfMute = data.self_mute;
        if ('self_stream' in data) this.streaming = data.self_stream;
        if ('self_video' in data) this.videoEnabled = data.self_video;
```

File: src/classes/VoiceState.ts [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+        return this;
+    }
+
+    /**
+     * Fetch the member from the Discord API.
+     * 
+     * @returns {Promise<VoiceState>} The updated voice state.
+     */
+    async fetch (): Promise<VoiceState> {
+        if (!this.member) this.member = await this.guild.members.fetch(this.memberId);
```
Comment (axisiscool) [inline src/classes/VoiceState.ts]:
```suggestion
        this.member ??= await this.guild.members.fetch(this.memberId);
```

**Review by axisiscool — CHANGES_REQUESTED** [review]

**Review by axisiscool — APPROVED** [review]

**Comment by knokbak** [comment]
chore: add methods to voice channel support (e.g. disconnect, setMute, setDeaf, move) @BirbJS/team

**Comment by PalmDevs** [comment]
anti stale!!!

**Comment by axisiscool** [comment]
@knokbak Any chance you can work on this?

**Pull request closed without merging**